### ISSUE TITLE
New room header claims to be simpler and more accessible but isn't

### ISSUE THREAD

**Issue by CobaltCause** [element-hq/element-web#27902]
### Steps to reproduce

1. Update to 1.11.72
2. Get taunted by this popup:
   ![image](https://github.com/user-attachments/assets/b58445f5-9579-4f93-b375-12f47be59d14)
4. Attempt to use the software

### Outcome

1. There is no obvious way to close the right sidebar after opening it anymore. There used to be an X button, but now there isn't. It seems like clicking anywhere in the room header will close the right sidebar when it's open, including clicking on:
   * "info" button
   * "threads" button
   * "people" button
   * Room avatar
   * Room name
   * Room topic
   * A link in the room topic, which, in addition to closing the sidebar, also opens the link (???)

   I assume all of these except the last one (I hope) are intended behavior. Seemingly the only exceptions are the voice and video call buttons. I think adding the dedicated X button back would be nice.
2. The room topic is now hidden by default; users must hover their cursor over the room header to reveal it. IMO this is very poor UX because there's often useful information in there and hiding it will make users forget this feature exists, leading to even more instances of e.g. people asking questions that would otherwise be answered by looking at the room topic. I've seen this happen where the user in question specifically pointed to this change as the reason they missed the room topic.
3. It's no longer possible to select text in the room topic.
4. If the room topic overflows the room header, it gets truncated. It is impossible to reveal the entire topic, as attempting to click on it will simply show/hide the right sidebar instead of presenting a dialog containing the full topic, like it used to do before the update.

   **Actually,** after messing with things for a few more minutes I realized that it *is* actually possible to view the entire room topic if it gets truncated in the room header. To do this, you must:
   * Open the right sidebar
   * Go to its "Info" tab
   * Notice that the room topic is also shown there in a truncated form
   * Guess that clicking on the truncated room topic in the right sidebar might do something
   * Click it, and it will indeed expand to show the full room topic
   
   It's also impossible to copy the text from this view of the room topic because the click will cause the topic to expand/collapse again which makes the selection go away.
5. If you click on one of the "info", "threads", or "people" buttons when the sidebar is closed, it will open that tab in the sidebar. If you then click any of the other icons than the one you clicked to open the sidebar, I would expect that it switch to that tab, but instead it closes the sidebar. (Clicking on the same button also closes the sidebar, but that at least makes some sense.)
6. The search button is no longer in the room header. It's tucked away under the right sidebar's "Info" tab at the very top, away from the other buttons, and has very low contrast so it's easy to miss. For example, you may notice that the previous version of this description simply said "The search button is gone." because I couldn't find it.
7. Because clicking most things in the room header now opens the right sidebar, the option to open the devtools for the current room by clicking the room icon or name in the room header is now gone. There is no devtools button in the right sidebar either. You must find the room in your (potentially large) list of rooms and right click on it there instead.

### Operating system

NixOS unstable

### Application version

Element desktop version 1.11.72

### How did you install the app?

nixpkgs

### Homeserver

computer.surgery

### Will you send logs?

No

**Comment by t3chguy** [comment]
1. https://github.com/element-hq/element-web/issues/27799
2. The topic is entirely gone from the header, that just hasn't been released yet, you can try it at develop.element.io - it lives in the right panel now.
3. It is meant to be possible in the right panel but there is a bug around it
4. See 2.
7. or you can use the quick settings (bottom left)

<img width="271" alt="image" src="https://github.com/user-attachments/assets/93b1c955-d0a5-4fb9-acd0-62ee4b52d7c3">

**Comment by CobaltCause** [comment]
> The topic is entirely gone from the header, that just hasn't been released yet, you can try it at develop.element.io - it lives in the right panel now.

... I really hope this gets reconsidered. In my experience, people often don't read room topics even when it's right in their face (i.e. viewable in the room timeline view with no extra clicks); I imagine hiding the topic behind extra mouse clicks and a closed-by-default sidebar will make this problem much worse.

**Comment by charliepark** [comment]
I find it baffling that search has been removed from the header. Searching past messages is one of the key tasks of a group chat tool; making it require several clicks to accomplish — especially in order to search all rooms, which is often the preferred default function — is a step backwards.

**Comment by jonpincus** [comment]
> I find it baffling that search has been removed from the header. Searching past messages is one of the key tasks of a group chat tool; making it require several clicks to accomplish — especially in order to search all rooms, which is often the preferred default function — is a step backwards.

I also find this baffling.  In fact I found this github thread when I asked in a Matrix chat room "what happened to the search box" and somebody pointed me here 🤣

**Comment by Benjamin-L** [comment]
> The topic is entirely gone from the header, that just hasn't been released yet, you can try it at develop.element.io - it lives in the right panel now.

Room topics are often very important to communicate about what discussion is on-topic for a given room, and what kinds of behavior are allowed. As a moderator in a relatively large room, removing the room topic from the header would be _highly_ undesirable for me, and I expect this change to result in an increase in moderation work. Ideally, I'd like it if the room topics were more prominent in the UI than they were before this change.

If it's likely to be taken into account for decision making, I could collect some statistics on how often people are redirected based on information visible in the room topic before and after this change.

**Comment by daniellekirkwood** [comment]
There's a lot of topics covered in this issue! I've taken a crack at breaking it out into sections to address all the feedback. Some of them point to other issues that we're working on.

> There is no obvious way to close the right sidebar after opening it anymore. 

We're working on this. There's a specific issue tracking it here: https://github.com/element-hq/element-web/issues/27799 
TLDR; Removing the tabs and introducing the [x] icon is likely the path forwards. And a change to how the buttons in the header act too. 


> The room topic is now hidden by default; users must hover their cursor over the room header to reveal it. 

The room topic is removed completely from the room header. It's available (expanded) in the right panel under the [i] icon. We have heard that the room topic changes are controversial, given there's lots of feedback on it, I've created an issue to collect the specific problems that y'all are experiencing and any ideas for moving forwards. https://github.com/element-hq/element-web/issues/27915

Apologies that there's lots of comments here that are valuable, I recognise it's not ideal to ask folks to comment _again_ however given how important this appears to be, I'd like to collect as much data as possible in one place so that our team can review it altogether.


> It's also impossible to copy the text from this view of the room topic because the click will cause the topic to expand/collapse again which makes the selection go away.

We'll take a look at this, thanks for raising!


> If you click on one of the "info", "threads", or "people" buttons when the sidebar is closed, it will open that tab in the sidebar. If you then click any of the other icons than the one you clicked to open the sidebar, I would expect that it switch to that tab, but instead it closes the sidebar.

We plan to change this as part of the re-introduction of the [x] icon (see above)


> The search button is no longer in the room header. It's tucked away under the right sidebar's "Info" tab at the very top, away from the other buttons, and has very low contrast so it's easy to miss. For example, you may notice that the previous version of this description simply said "The search button is gone." because I couldn't find it.

🤔  We'll double check the contrast on this field - it should pass accessibility checks but thanks for raising. It should also be the full width of the right panel. We plan to address Search holistically in the future.


> Because clicking most things in the room header now opens the right sidebar, the option to open the devtools for the current room by clicking the room icon or name in the room header is now gone. There is no devtools button in the right sidebar either. You must find the room in your (potentially large) list of rooms and right click on it there instead.

Thanks! Do you think it's necessary that we add DevTools into the options in the right panel? 
Note that DevTools can be accessed for a room by typing "/Devtools" into the composer and via Search (CMD+K, _type room name_, click 3 dots > developer tools

**Comment by CobaltCause** [comment]
> Removing the tabs and introducing the [x] icon is likely the path forwards. And a change to how the buttons in the header act too.

This sounds much more intuitive and less cluttered to me. Would also fix the open/close/switch confusion as you noted later your comment at the design level which is good.

> Do you think it's necessary that we add DevTools into the options in the right panel?

I don't think that's necessary, no; mainly because of the alternative pointed out in <https://github.com/element-hq/element-web/issues/27902#issuecomment-2270662660>.

> We plan to address Search holistically in the future.

I hope this includes putting the search function on display at all times somewhere in the room view (such as the header like how it was before), as putting it inside something else feels too hidden IMO. I'm curious what was wrong with the previous design such that it had to be changed in the first place.

**Comment by jeffrson** [comment]
> 6\. The search button is no longer in the room header. It's tucked away under the right sidebar's "Info" tab at the very top, away from the other buttons, and has very low contrast so it's easy to miss. For example, you may notice that the previous version of this description simply said "The search button is gone." because I couldn't find it.

Really, that's a problem. Why was it changed? And the channel search still recommends clicking the magnifying glass icon to search for messages.

**Comment by t3chguy** [comment]
1, 3, 4, 5 are fixed

2 is https://github.com/element-hq/element-web/issues/27915
6 is intentional bar the contrast which is being tackled in separate streams of work
7 as already answered devtools lives in Quick Settings too and thus intentional

**Comment by t3chguy** [comment]
That isn't being changed, only the contrast issue as part of a larger accessibility task.

**Comment by Xiretza** [comment]
Given how well the search works, maybe it's for the best that nobody will be able to find it anymore going forward.

**Comment by lheckemann** [comment]
Wait, so making search more complicated to access is intentional? I don't want to be combative, but... why?
